backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -890,26 +924,31 @@ trait StandardAsyncExecutionActor extends AsyncBackendJobExecutionActor with Sta
         case (stderrSize, returnCodeAsString) =>
           val tryReturnCodeAsInt = Try(returnCodeAsString.trim.toInt)
 
-          if (isSuccess(status)) {
+          if (isDone(status)) {
             tryReturnCodeAsInt match {
               case Success(returnCodeAsInt) if failOnStdErr && stderrSize.intValue > 0 =>
-                Future.successful(FailedNonRetryableExecutionHandle(StderrNonEmpty(jobDescriptor.key.tag, stderrSize, stderrAsOption), Option(returnCodeAsInt)))
+                retryElseFail(status, FailedNonRetryableExecutionHandle(StderrNonEmpty(jobDescriptor.key.tag, stderrSize, stderrAsOption), Option(returnCodeAsInt)))
               case Success(returnCodeAsInt) if isAbort(returnCodeAsInt) =>
                 Future.successful(AbortedExecutionHandle)
               case Success(returnCodeAsInt) if !continueOnReturnCode.continueFor(returnCodeAsInt) =>
-                Future.successful(FailedNonRetryableExecutionHandle(WrongReturnCode(jobDescriptor.key.tag, returnCodeAsInt, stderrAsOption), Option(returnCodeAsInt)))
+                retryElseFail(status, FailedNonRetryableExecutionHandle(WrongReturnCode(jobDescriptor.key.tag, returnCodeAsInt, stderrAsOption), Option(returnCodeAsInt)))
               case Success(returnCodeAsInt) =>
                 handleExecutionSuccess(status, oldHandle, returnCodeAsInt)
               case Failure(_) =>
+                //What does it mean that there is no return code-- should this be retried??
```

Comment:
It means for some reason the return code couldn't be parsed from String to Int for some reason, which would likely mean we messed up the Cromwell script. I don't thing this should be retried